lib/std/fmt.zig

```diff
@@ -1448,6 +1455,7 @@ fn parseWithSign(
     var x: T = 0;
 
     for (buf_start) |c| {
+        if (c == '_') continue;
```

Comment:
This is not enough as it will happily parse `_123` and `123_`, you need a check (and tests) to make sure the character is not the first nor the last in `buf_start`.